pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2011,6 +2021,13 @@ type ObjectStoreUserSpec struct {
 	// +optional
 	// +nullable
 	Quotas *ObjectUserQuotaSpec `json:"quotas,omitempty"`
+	// Allows specifying credentials for the user. If not provided, the operator
+	// will generate them.
+	// +optional
+	Keys []ObjectUserKey `json:"keys,omitempty"`
+	// Disable creation of the automatic secret with S3 credentials.
+	// +optional
+	DisableAutomaticSecret bool `json:"disableAutomaticSecret,omitempty"`
```

Comment:
IMO, this isn't really needed or useful. If users don't want to use the generated secret, we aren't forcing them. Can we remove this please?

Reply:
As I explained in this [comment](https://github.com/rook/rook/pull/15359#discussion_r2006413808), I actively do not want the automatically generated secrets when explicit keys are set.  At best, in that scenario the automatic secrets are clutter  when running `k -n rook-ceph get secrets` and I feel there is a high chance they will end up being a foot gun for someone to manually modify with the expectation that it will update the keys set on the rgw user.  My preference was the first iteration where setting explicit keys caused the automatic secret to be removed.  Adding a flag here is a compromise.  The integration tests cover transitioning back forth between explicit keys and not both with and without `DisableAutomaticSecret` set.

Reply:
Having a flag is worse IMO, just CR clutter. Can we please remove changes to this secret's management from this PR and follow up later?

Reply:
Done.